### PR TITLE
Format computeNewTimeStep

### PR DESCRIPTION
**Pull request by henry2004y**
Format and return type changes for the time step calculation.

### CODE REVIEW

**Review by markusbattarbee — CHANGES_REQUESTED** [review]
Good call on changing the return type. Perhaps a few other changes as well now that we're at it?

File: vlasiator.cpp [hunk 1]
```diff
@@ -108,156 +107,127 @@ bool computeNewTimeStep(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpi
...
+      if (cell->sysBoundaryFlag == sysboundarytype::NOT_SYSBOUNDARY ||
+          (cell->sysBoundaryLayer == 1 && cell->sysBoundaryFlag != sysboundarytype::NOT_SYSBOUNDARY)) {
+         // spatial fluxes computed also for boundary cells
+         dtMaxLocal[0] = min(dtMaxLocal[0], cell->parameters[CellParams::MAXRDT]);
       }
 
       if (cell->sysBoundaryFlag == sysboundarytype::NOT_SYSBOUNDARY && cell->parameters[CellParams::MAXVDT] != 0) {
-         //Acceleration only done on non-boundary cells
-         dtMaxLocal[1]=min(dtMaxLocal[1], cell->parameters[CellParams::MAXVDT]);
+         // acceleration only done on non-boundary cells
```
[the remark this answers is not in the document]
Comment (henry2004y) [reply vlasiator.cpp:154]: Do you mean something like

```cpp
if (P::vlasovAccelerateMaxwellianBoundaries && cell->sysBoundaryFlag == sysboundarytype::SET_MAXWELLIAN) {
   dtMaxLocal[1] = min(dtMaxLocal[1], cell->parameters[CellParams::MAXVDT]);
}
```
?

[the remark this answers is not in the document]
Comment (markusbattarbee) [reply vlasiator.cpp:154]: Yeah, perhaps also the check for `cell->parameters[CellParams::MAXVDT] != 0` just to be sure

File: vlasiator.cpp [hunk 2]
```diff
@@ -137,11 +137,11 @@ void computeNewTimeStep(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpi
                    blockParams[blockLID * BlockParams::N_VELOCITY_BLOCK_PARAMS + BlockParams::VZCRD] +
                    (i + HALF) * blockParams[blockLID * BlockParams::N_VELOCITY_BLOCK_PARAMS + BlockParams::DVZ] + EPS;
 
-               const Real dt_max_cell = min(dx / fabs(Vx), min(dy / fabs(Vy), dz / fabs(Vz)));
-               cell->parameters[CellParams::MAXRDT] = min(dt_max_cell, cell->parameters[CellParams::MAXRDT]);
+               const Real dt_max_cell = min({dx / fabs(Vx), dy / fabs(Vy), dz / fabs(Vz)});
                cell->set_max_r_dt(popID, min(dt_max_cell, cell->get_max_r_dt(popID)));
             }
          }
+         cell->parameters[CellParams::MAXRDT] = min(cell->get_max_r_dt(popID), cell->parameters[CellParams::MAXRDT]);
```
Comment (markusbattarbee) [inline vlasiator.cpp:144]: still needs to be inside the pop loop!

File: vlasiator.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (henry2004y) [reply vlasiator.cpp:156]: Hopefully this is better.

File: vlasiator.cpp (hunk 2 above)
Comment (henry2004y) [reply vlasiator.cpp:144]: I think it is, otherwise it won't compile?

Comment (markusbattarbee) [reply vlasiator.cpp:144]: Sorry, my bad :) For some reason even looking at it multiple times, I didn't see the i loop until going out of diff mode.

**Comment by ursg** [comment]
Yup, this looks nice.
When setting the github diff viewer to ignore whitespace changes, there's actually very little in terms of semantic change here.
Good to merge into dev IMHO, but I'll leave it to you to push the button, Markus. :)

**Pull request merged**